core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala

```diff
@@ -54,14 +54,28 @@ object ApplicationDescriptor {
       blueprint: VerifiedBlueprint,
       agentPaths: Map[String, String],
       libraryVersion: String): ApplicationDescriptor = {
+    apply(appId, appVersion, (_: String) => image, blueprint, agentPaths, libraryVersion)
+  }
+
+  def apply(
+      appId: String,
+      appVersion: String,
+      images: String => String,
```

Comment:
?

Reply:
Is just more generic: I can implement this method with `Map[String, String].apply(key)` as well as a `TypesafeConfig.getString(key)`

Reply:
minor, would prefer a name like `imageSelector` or `getImageByKey` or something like that